hail/src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -261,12 +261,20 @@ class IRSuite extends HailSuite {
     assertEvalsTo(Coalesce(FastSeq(In(0, TInt32()))), FastIndexedSeq((null, TInt32())), null)
     assertEvalsTo(Coalesce(FastSeq(In(0, TInt32()))), FastIndexedSeq((1, TInt32())), 1)
     assertEvalsTo(Coalesce(FastSeq(NA(TInt32()), In(0, TInt32()))), FastIndexedSeq((null, TInt32())), null)
-    assertEvalsTo(Coalesce(FastSeq(NA(TInt32()), In(0, TInt32()))), FastIndexedSeq((1, TInt32())), 1)
+    assertEvalsTo(Coalesce(FastSeq(NA(TInt32()), In(0, TInt32(true)))), FastIndexedSeq((1, TInt32(true))), 1)
```

Comment:
To test difference in requiredeness of scalars, ensure we're not relying on it. Of course this will go away once we remove requiredeness of Type, but for now helpful to test the Emit change.

Reply:
We'll make sure we test this when requiredness leaves virtual type by removing the `In()` constructor that takes a virtual type.

Reply:
https://github.com/hail-is/hail/issues/7973